src/main/java/org/mcmmo/mcmmotowny/listeners/ExperienceListener.java

```diff
@@ -36,4 +42,25 @@ private boolean isAffectedSkill(String skillName) {
     private boolean isAffectedReason(String xpGainReason) {
         return Config.getInstance().getAffectedXpGainReasons().contains(xpGainReason);
     }
+
+    @EventHandler
+    public void onHardcoreDeath(McMMOPlayerDeathPenaltyEvent event) {
+        TownBlock block = TownyAPI.getInstance().getTownBlock(event.getPlayer().getLocation());
+        if (block != null && block.getType().equals(TownBlockType.ARENA))
+            event.setCancelled(true);
+    }
+
+    @EventHandler
+    public void onStatLoss(McMMOPlayerStatLossEvent event) {
+        TownBlock block = TownyAPI.getInstance().getTownBlock(event.getPlayer().getLocation());
+        if (block != null && block.getType().equals(TownBlockType.ARENA))
+            event.setCancelled(true);
+    }
+
+    @EventHandler
+    public void onVampirism(McMMOPlayerVampirismEvent event) {
+        TownBlock block = TownyAPI.getInstance().getTownBlock(event.getPlayer().getLocation());
```

Comment:
Instead of checking blocks you should be doing a bounds check within the region, what do you think?

Reply:
Not sure what you mean by a bounds check, is that something in the McMMO api?

Reply:
Ah I think I understand the miscommunication: the units Towny uses as regions are TownBlocks, that API call gets the TownBlock that the player is located in.

Reply:
Does towny offer any inBounds() checks for its regions? If not, then I suppose I won't have you go through the trouble of figuring out how to do that yourself.

Reply:
The getTownBlock() call is getting the region the player is in, Towny's "regions" are made up of (usually) 16 x 16 x World-Height delimited TownBlocks. The code above is getting the TownBlock that the player is stood in, make sure its not null (this would mean they're in the Wilderness outside of any Town,) and then is it finding out if the TownBlock has a type of Arena.